R/bayesiannetworkanalysis.R

```diff
@@ -0,0 +1,1297 @@
+    dataset <- Reduce(rbind.data.frame, dataset)
+    
+    if (options[["groupingVariable"]] != "") {
+      # these cannot be chained unfortunately
+      groupingVariableName <- options[["groupingVariable"]]
+      dfGroup <- data.frame(groupingVariable)
+      colnames(dfGroup) <- .v(groupingVariableName)
+      .hasErrors(dataset = dfGroup,
+                 type = c("missingValues", "factorLevels"),
+                 missingValues.target = groupingVariableName,
+                 factorLevels.target = groupingVariableName,
+                 factorLevels.amount = "< 2",
+                 exitAnalysisIfErrors = TRUE)
+      dataset[[.v(options[["groupingVariable"]])]] <- groupingVariable
+      groupingVariable <- options[["groupingVariable"]]
+    }
+  }
+}
+
+.bayesianNetworkAnalysisRun <- function(mainContainer, dataset, options) {
+  
+  # List that contains state or is empty:
+  networkList <- list(
+    network    = mainContainer[["networkState"]]$object,
+    centrality = mainContainer[["centralityState"]]$object,
+    layout     = mainContainer[["layoutState"]]$object
+  )
+  
+  if (length(options[["variables"]]) <= 2L)
+    return(networkList)
+  
+  if (is.null(networkList[["network"]]))
+    tryCatch(
+      networkList[["network"]] <- .bayesianNetworkAnalysisComputeNetworks(options, dataset),
+      error = function(e) {
+        mainContainer$setError(.extractErrorMessage(e))
+      }
+    )
+  
+  if (!mainContainer$getError() && !is.null(networkList[["network"]])) { 
+    if (is.null(networkList[["layout"]]))
+      networkList[["layout"]] <- .bayesianNetworkAnalysisComputeLayout(networkList[["network"]], dataset, options)
+    
+    if (is.null(networkList[["centrality"]]) && (options[["tableCentrality"]] || options[["plotCentrality"]]))
+      networkList[["centrality"]] <- .bayesianNetworkAnalysisComputeCentrality(networkList[["network"]])
+    
+    names(networkList[["network"]]) <- names(dataset) # <- names(networkList[["centrality"]])
+    
+    mainContainer[["networkState"]]    <- createJaspState(networkList[["network"]])
+    mainContainer[["centralityState"]] <- createJaspState(networkList[["centrality"]], dependencies = c("maxEdgeStrength", "minEdgeStrength"))
+    mainContainer[["layoutState"]]     <- createJaspState(networkList[["layout"]], 
+                                                          dependencies = c("layout", "repulsion", "layoutX", "layoutY"))
+    
+    #.networkAnalysisSaveLayout(mainContainer, options, networkList[["layout"]])
+  }
+  
+  return(networkList)
+}
+
+.bayesianNetworkAnalysisComputeCentrality <- function(networks) {
+  
+  centralities <- vector("list", length(networks))
+  for (nw in seq_along(networks)) {
+    
+    network <- networks[[nw]]
+    
+    # add measures 
+    centralitySamples <- centrality(network = network)
+    
+    nSamples <- nrow(network$samplesPosterior)
+    
+    value <- apply(centralitySamples[, 3:(nSamples+2)], MARGIN = 1, mean)
```

Comment:
```suggestion
	posteriorMeans <- colMeans(centralitySamples[, 3:(nSamples+2)])
```
- A slight preference for an informative name (e.g., `posteriorMeans`) over a generic name (e.g., `value`). 
- in general, `colMeans` (and `rowMeans`) are more efficient than `apply(x, 2L, mean)`. 
- could you add a comment to explain why we're doing `3:(nSamples+2)`?